integration-tests/support/pages/tabs/PipelinerunsTabPage.ts

```diff
@@ -0,0 +1,116 @@
+export class DetailsTab {
+    static goToDetailsTab() {
+        cy.get(pipelinerunsTabPO.clickDetailsTab).click();
+    }
+
+    static checkStatusSucceeded(isAdvancedFlowActive: boolean = false) {
+        // //cy.wait(5000);
+        // cy.get(pipelinerunsTabPO.statusPO, { timeout: 5000 }).invoke('text').then(text => {
+        //     if (text.includes('Running')) {
+        //         cy.get(pipelinerunsTabPO.statusPO, { timeout: 720000 }).should('not.have.text', 'Running');
+        //     }
+
+        //     if (text.includes('Failed')) {
+        //         LogsTab.goToLogsTab();
+        //         cy.screenshot('capture-logs-on-pipelinerun-failure', {capture: 'fullPage'});
+        //     }
+        // })
+
+        //cy.wait(25000);
+        cy.get(pipelinerunsTabPO.statusPO, { timeout: 25000 }).invoke('text').then(text => {
+            if (text.includes('Running')) {
+                cy.get(pipelinerunsTabPO.statusPO, { timeout: 900000 }).should('not.have.text', 'Running');
+            }
+        });
+
+        cy.get(pipelinerunsTabPO.statusPO).invoke('text').then(text => {
+            if (text.includes('Succeeded')) {
+                TaskRunsTab.goToTaskRunsTab().assertTaskNamesAndTaskRunStatus(isAdvancedFlowActive);
+            }
+            else if (text.includes('Failed')) {
+                LogsTab.goToLogsTab();
+                cy.screenshot('capture-logs-on-pipelinerun-failure', {capture: 'fullPage'});
+            }
+        });
+    }
+}
+
+export class TaskRunsTab {
+    taskNameList: string[] = ['init', 'git-clone', 'configure-build', 'buildah', 'summary',
+                                    'sast-snyk-check', 'deprecated-image-check','clamav-scan',
+                                    'sbom-json-check', 'sanity-inspect-image', 'clair-scan',
+                                    'sanity-label-check', 'sanity-label-check'];
+
+    static goToTaskRunsTab() {
+        cy.get(pipelinerunsTabPO.clickTaskRunsTab).click();
+        return new TaskRunsTab();
+    }
+
+    assertTaskNamesAndTaskRunStatus(isAdvancedFlowActive: boolean = false) {
+        for (let i = 0; i < ((isAdvancedFlowActive)? this.taskNameList.length : 5); i++) {
+            cy.get(`[data-test="${i}-0"] > td`).eq(1).then((taskName) => {
+                expect(this.taskNameList.includes(taskName.text().trim())).to.equal(true);
+            });
+
+            cy.get(`[data-test="${i}-0"] > td`).eq(3).within(() => {
+                cy.get(pipelinerunsTabPO.taskRunStatus).invoke('text').then(status => {
+                    if (status.includes('Failed')) {
+                        cy.screenshot('taskrun-failed-even-though-pipelinerun-succeeded', {capture: 'fullPage'});
+                        return;
+                    }
+                    else if (status.includes('Pending')) {
+                        cy.screenshot('taskrun-pending-even-though-pipelinerun-succeeded', {capture: 'fullPage'});
+                        return;
+                    }
```

Comment:
shouldn't the test just fail if the pipeline doesn't succeed? in which case the screenshot would be taken automatically

Reply:
makes sense. done, thanks